src/cloudsc_cuf_loki/cuf_cloudsc_driver_loki_mod.F90

```diff
@@ -0,0 +1,200 @@
+    REAL(KIND=JPRB),    INTENT(IN)    :: PLU(NPROMA,NLEV,NGPBLKS)   ! Conv. condensate
+    REAL(KIND=JPRB),    INTENT(INOUT) :: PLUDE(NPROMA,NLEV,NGPBLKS) ! Conv. detrained water
+    REAL(KIND=JPRB),    INTENT(IN)    :: PSNDE(NPROMA,NLEV,NGPBLKS) ! Conv. detrained snow
+    REAL(KIND=JPRB),    INTENT(IN)    :: PMFU(NPROMA,NLEV,NGPBLKS)  ! Conv. mass flux up
+    REAL(KIND=JPRB),    INTENT(IN)    :: PMFD(NPROMA,NLEV,NGPBLKS)  ! Conv. mass flux down
+    REAL(KIND=JPRB),    INTENT(IN)    :: PA(NPROMA,NLEV,NGPBLKS)    ! Original Cloud fraction (t)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PCLV(NPROMA,NLEV,NCLV,NGPBLKS)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PSUPSAT(NPROMA,NLEV,NGPBLKS)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PLCRIT_AER(NPROMA,NLEV,NGPBLKS)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PICRIT_AER(NPROMA,NLEV,NGPBLKS)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PRE_ICE(NPROMA,NLEV,NGPBLKS)
+    REAL(KIND=JPRB),    INTENT(IN)    :: PCCN(NPROMA,NLEV,NGPBLKS)     ! liquid cloud condensation nuclei
+    REAL(KIND=JPRB),    INTENT(IN)    :: PNICE(NPROMA,NLEV,NGPBLKS)    ! ice number concentration (cf. CCN)
+
+    REAL(KIND=JPRB),    INTENT(INOUT) :: PCOVPTOT(NPROMA,NLEV,NGPBLKS) ! Precip fraction
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PRAINFRAC_TOPRFZ(NPROMA,NGPBLKS)
+    ! Flux diagnostics for DDH budget
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQLF(NPROMA,NLEV+1,NGPBLKS)  ! Flux of liquid
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQIF(NPROMA,NLEV+1,NGPBLKS)  ! Flux of ice
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFCQLNG(NPROMA,NLEV+1,NGPBLKS) ! -ve corr for liq
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFCQNNG(NPROMA,NLEV+1,NGPBLKS) ! -ve corr for ice
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQRF(NPROMA,NLEV+1,NGPBLKS)  ! Flux diagnostics
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQSF(NPROMA,NLEV+1,NGPBLKS)  !    for DDH, generic
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFCQRNG(NPROMA,NLEV+1,NGPBLKS) ! rain
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFCQSNG(NPROMA,NLEV+1,NGPBLKS) ! snow
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQLTUR(NPROMA,NLEV+1,NGPBLKS) ! liquid flux due to VDF
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFSQITUR(NPROMA,NLEV+1,NGPBLKS) ! ice flux due to VDF
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFPLSL(NPROMA,NLEV+1,NGPBLKS) ! liq+rain sedim flux
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFPLSN(NPROMA,NLEV+1,NGPBLKS) ! ice+snow sedim flux
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFHPSL(NPROMA,NLEV+1,NGPBLKS) ! Enthalpy flux for liq
+    REAL(KIND=JPRB),    INTENT(OUT)   :: PFHPSN(NPROMA,NLEV+1,NGPBLKS) ! Enthalp flux for ice
+
+    INTEGER(KIND=JPIM) :: JKGLO,IBL,ICEND
+
+    ! Local copy of cloud parameters for offload
+    !TYPE(TECLDP) :: LOCAL_YRECLDP
+
+    TYPE(PERFORMANCE_TIMER) :: TIMER
+    INTEGER(KIND=JPIM) :: TID ! thread id from 0 .. NUMOMP - 1
+
+    IBL = 1  ! Useless statement to show the compiler that the sepcification part is over!
+
+    !@cuf CALL YOMCST_UPDATE_DEVICE()
```

Comment:
These, as well as the treatment seem to be the only current deviations from the base-line `cloudsc_loki` variant. This suggest we should be able to get consolidated before merging this, no?